planning/mission_planner/src/lanelet2_plugins/default_planner.cpp

```diff
@@ -180,21 +189,63 @@ PlannerPlugin::MarkerArray DefaultPlanner::visualize(const HADMapRoute & route)
   return route_marker_array;
 }
 
-bool DefaultPlanner::is_goal_valid(const geometry_msgs::msg::Pose & goal) const
+visualization_msgs::msg::MarkerArray DefaultPlanner::visualize_debug_footprint(
+  tier4_autoware_utils::LinearRing2d goal_footprint_) const
+{
+  visualization_msgs::msg::MarkerArray msg;
+  auto marker = tier4_autoware_utils::createDefaultMarker(
+    "map", rclcpp::Clock().now(), "goal_footprint", 0, visualization_msgs::msg::Marker::LINE_STRIP,
+    tier4_autoware_utils::createMarkerScale(0.05, 0.0, 0.0),
+    tier4_autoware_utils::createMarkerColor(0.99, 0.99, 0.2, 1.0));
+  marker.lifetime = rclcpp::Duration::from_seconds(2.5);
+
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint_[0][0], goal_footprint_[0][1], 0.0));
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint_[1][0], goal_footprint_[1][1], 0.0));
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint_[2][0], goal_footprint_[2][1], 0.0));
+  marker.points.push_back(
+    tier4_autoware_utils::createPoint(goal_footprint_[3][0], goal_footprint_[3][1], 0.0));
+  marker.points.push_back(marker.points.front());
+
+  msg.markers.push_back(marker);
+
+  return msg;
+}
+
+bool DefaultPlanner::is_goal_valid(const geometry_msgs::msg::Pose & goal)
 {
+  const auto logger = node_->get_logger();
   lanelet::Lanelet closest_lanelet;
   if (!lanelet::utils::query::getClosestLanelet(road_lanelets_, goal, &closest_lanelet)) {
     return false;
   }
+
+  const auto local_vehicle_footprint = createVehicleFootprint(vehicle_info_);
+  tier4_autoware_utils::LinearRing2d goal_footprint_ =
+    transformVector(local_vehicle_footprint, tier4_autoware_utils::pose2transform(goal));
+
+  // check if goal footprint exceeds lane
+  std::vector<tier4_autoware_utils::Point2d> points_intersection;
+  boost::geometry::intersection(
+    goal_footprint_, closest_lanelet.polygon2d().basicPolygon(), points_intersection);
```

Comment:
@takayuki5168 @TakaHoribe @mitsudome-r 
The closest may not be enough at the boundary of two lanelets. I'm not familiar with lanelet so I'd like to hear other comments.

Reply:
I am planning to update that part as follows: Combining the next lanelets of the goal lanelet with the the previous lanes on the route. Then, i can check that polygon with goal footprint.